scripts/create-new-spec.js

```diff
@@ -1,23 +1,48 @@
+const getSpecVersion = require('./get-spec-version');
+const updateExistingSpec = require('./update-existing-spec');
 const sdk = require('api')('@developers/v2.0#kpjtacsldjbscf9');
+const fs = require('fs');
 
-module.exports = async function createNewSpec(apiKey, spec_file,version_number) {
-  try{
+module.exports = async function createNewSpec(apiKey, spec_file, version_number, filePath) {
+  try {
     await sdk.auth(apiKey);
-    let created_doc =  await sdk
-    .uploadAPISpecification(
+    console.log('[INFO] inside createNewSpec function with version number ', version_number);
+    let created_doc = await sdk.uploadAPISpecification(
       {
         spec: spec_file,
       },
       {
         accept: 'application/json',
-        'x-readme-version': version_number
+        'x-readme-version': version_number,
       },
     );
-    console.log("[INFO] Doc created ",created_doc);
+    console.log('[INFO] Doc created ', created_doc);
     return created_doc;
   } catch (err) {
-    var message = await err.json();
-    console.error("[ERROR] Error creating new spec.",message); 
-    throw new Error(message);
+    if (err.status == 503) {
+      console.log('[INFO] 503 error, trying again');
```

Comment:
Added extra logic here to support retry with the update api of readme after finding the latest version if it exists.